site/content/docs/main/csi-snapshot-data-movement.md

```diff
@@ -0,0 +1,391 @@
+
+- CSI plugin expects a data mover to handle the `DataUpload` CR. If no data mover is configured for the backup, Velero built-in data mover will handle it.  
+- If the `DataUpload` CR does not reach to the terminal state with in the given time, the `DataUpload` CR will be cancelled. You can set the timeout value per backup through the `--item-operation-timeout` parameter, the default value is `4 hours`.  
+
+- Velero built-in data mover creates a volume from the CSI snapshot and transfer the data to the backup storage according to the backup storage location defined by users.  
+- After the volume is created from the CSI snapshot, Velero built-in data mover waits for Kubernetes to provision the volume, this may take some time varying from storage providers, but if the provision cannot be finished in a given time, Velero built-in data mover will cancel this `DataUpload` CR. The timeout is configurable through a node-agent's parameter `data-mover-prepare-timeout`, the default value is 30 minutes.  
+- When the data transfer completes or any error happens, Velero built-in data mover sets the `DataUpload` CR to the terminal state, either `Completed` or `Failed`.  
+- Velero built-in data mover also monitors the cancellation request to the `DataUpload` CR, once that happens, it cancels its ongoing activities, cleans up the intermediate resources and set the `DataUpload` CR to `Cancelled`.  
+
+### Restore
+
+Velero restores resources for CSI snapshot data movement restore in the same way as other restore types. When it encounters a PVC, particular logics will be conducted: 
+
+- When it finds a PVC object, Velero calls CSI plugin through a Restore Item Action.  
+- CSI plugin checks the backup information, if a data movement was involved, it creates a `DataDownload` CR and then returns to Velero restore.  
+- Velero is now able to restore other resources, including other PVC objects.  
+- Velero restore periodically queries the data movement status from CSI plugin, the period is configurable through the Velero server parameter `--item-operation-sync-frequency`, by default it is 10s. On the call, CSI plugin turns to check the phase of the `DataDownload` CRs.  
+- When all `DataDownload` CRs come to a terminal state (i.e., `Completed`, `Failed` or `Cancelled`), Velero restore will finish.  
+
+- CSI plugin expects the same data mover for the backup to handle the `DataDownload` CR. If no data mover was configured for the backup, Velero built-in data mover will handle it.  
+- If the `DataDownload` CR does not reach to the terminal state with in the given time, the `DataDownload` CR will be cancelled. You can set the timeout value per backup through the same `--item-operation-timeout` parameter.  
+
+- Velero built-in data mover creates a volume with the same specification of the source volume.  
```

Comment:
Shall we stress that the volume creation is done via dynamic provision, i.e. it only creates the PVC and it has dependency on storageclass on the target cluster?

Reply:
We have mentioned how to configure the storage class on the target in section `Configure storage class on target cluster`.
I have rephrased the sentence in the same section as below:
`On the other hand, Velero built-in data movement creates a PVC with the same specification as it in the source cluster and expects the volume to be provisioned similarly to the source cluster. For example, the same storage class should be working in the target cluster`